### ISSUE TITLE
FERC plant records showing up in multiple time series

### ISSUE THREAD

**Issue by zaneselvans** [catalyst-cooperative/pudl#240]
For some reason a small fraction of the FERC plant records are showing up as associated with more than one FERC plant ID.  This should never happen, and so it means something is broken! Deep down inside the `where_matches()` and `best_matches()` functions.

This is a sub-issue of #144

**Comment by zaneselvans** [comment]
The problem here is in the way the plant_id_ferc1 values are assigned. Right now we're considering a time series valid if every record within it only shows up in time series which are mutually consistent -- some of them may have gaps (depending on which year is used as the seed for matching) but none of them have *other* records that are in conflict with each other. This flexibility dramatically increases the number of records which are successfully assigned to a time series, but it makes assigning the ID a little more complicated, since we need to need to ensure that any record that shows up in any one of these potentially partial time series gets assigned the same plant ID. This means that it's the union of all records that are found to be associated with each other that make up the time series that gets a plant ID.

**Comment by zaneselvans** [comment]
I've got this more complicated assignment of plant_ids working now, in the same way that we did the generation unit assignment using connections between boilers and generators -- creating a graph and finding the connected components, each of which gets assigned a FERC plant ID.  But there's still something fishy going on. Every single record_id should *either* end up assigned to a FERC plant_id in the main graph, *or* it should end up orphaned, and get assigned a FERC plant_id after the fact.  But for some reason, there are still 91 records that end up without a plant ID.  Still debugging.
